### ISSUE TITLE
Decode UUID type

### ISSUE THREAD

**Issue by ysbaddaden** [will/crystal-pg#20]
Querying UUID columns results in an invalid UTF-8 String. I suppose it's missing a proper decoder (oid = 2950)?

What is weird is that this started failing when I upgraded to crystal v0.8.0 and crystal-pg v0.4.0. It used to work previously (maybe crystal 0.7.7 + pg v0.3.2).

**Comment by ysbaddaden** [comment]
This correctly decodes the UUID:

```crystal
class UuidDecoder < Decoder
  def decode(value_ptr)
    value_ptr.to_slice(16).hexstring
  end
end

register_decoder UuidDecoder.new, 2950
```

**Comment by will** [comment]
Thanks. The 0.4 release switched to the binary result protocol from the text one. There have been a few cases of types that I missed having to do something special about, and it looks like this is another one.

**Comment by ysbaddaden** [comment]
Sweet :-)

**Comment by will** [comment]
Also released as 0.4.1
